### PR TITLE
Adds shape metadata manipulation with tag helpers

### PR DESCRIPTION
**Pull request by paynecrl97**
For #1227

### CODE REVIEW

File: src/OrchardCore/OrchardCore.DisplayManagement/TagHelpers/BaseShapeTagHelper.cs [hunk 1]
```diff
@@ -108,6 +111,8 @@ protected BaseShapeTagHelper(IShapeFactory shapeFactory, IDisplayHelperFactory d
                 }
             }
 
+            await output.GetChildContentAsync();
```
Comment (paynecrl97) [inline src/OrchardCore/OrchardCore.DisplayManagement/TagHelpers/BaseShapeTagHelper.cs]: This method is annotated with `MustUseReturnValue`, but I'm not sure why, or how I should be using it?

Additionally- `SetHtmlContent` called immediately after this line is also annotated with this attribute, but the return value is also being ignored here.

**Comment by sebastienros** [comment]
Why using another tag `<metadata>` ?
Is it just to have a way to set the common properties? then what about moving it to `BaseTagHelper`? 
Would be nice also to be able to set the classes and id the same way. Not sure it's already supported, it would just update the IShape properties.

**Comment by paynecrl97** [comment]
The `metadata` section serves a few purposes:

- It allows the user to set properties on the `Metadata` object itself e.g. `<metadata display-type="summary" />`
- It maintains familiarity for the user because the tags match how you'd do it in C# e.g. `myShape.Metadata.AddAlternate("my-alternate");`
- It allows the child tags to set the `ParentTag` property on the `HtmlTargetElementAttribute` which enables intellisense for these tags. e.g. https://github.com/OrchardCMS/OrchardCore/pull/1228/files#diff-ac30c0e3c44407456a5905b9c089002bR8

File: src/OrchardCore/OrchardCore.DisplayManagement/TagHelpers/AddAlternateTagHelper.cs [hunk 2]
```diff
@@ -10,9 +9,9 @@ public class AddAlternateTagHelper : TagHelper
     {
         public override Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
         {
-            var metadata = (ShapeMetadata)context.Items[typeof(ShapeMetadata)];
+            var shape = (IShape)context.Items[typeof(IShape)];
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.DisplayManagement/TagHelpers/AddAlternateTagHelper.cs]: if (context.Items[typeof(IShape)] as IShape shape)
{
  shape.Metadata.Alternates.Add(Convert.ToString(output.Attributes["name"].Value));
}

**Pull request merged**